backend/src/main/kotlin/cz/loono/backend/api/service/HealthcareProvidersService.kt

```diff
@@ -76,13 +93,19 @@ class HealthcareProvidersService @Autowired constructor(
         serverPropertiesRepository.save(firstProperties)
     }
 
-    private fun updateCache() {
+    @Synchronized
+    fun updateCache() {
         providersCache.invalidateAll()
         fileCache.invalidateAll()
-        val providers = healthcareProviderRepository.findAll()
-        providers.forEach { provider ->
-            val id = HealthcareProviderIdDto(locationId = provider.locationId, institutionId = provider.institutionId)
-            providersCache.put(id, provider)
+        val providers = mutableSetOf<HealthcareProvider>()
```

Comment:
Podľa výstupu z jfr je toto jeden z problémov. Skús inicializovať set podľa očakávanej veľkosti.

Reply:
to je zajimavy tip, rovnou ten set vytvorit v dane velikosti. muzeme zkusit.